src/proto/FunctionRpc.proto

```diff
@@ -260,6 +264,30 @@ message RpcFunctionMetadata {
 
   // Is set to true for proxy
   bool is_proxy = 7;
+
+  // Function indexing status
+  StatusResult status = 8;
+
+  // Function language
+  string language = 9;
+
+  // Raw binding info
+  repeated string raw_bindings = 10;
+}
+
+// Host tells worker it is ready to receive metadata
+message FunctionsMetadataRequest {
+  // base directory for function app
+  string directory = 1;
```

Comment:
Is this message needed to enforce the order of communication? If yes, rename to directory to `function_app_directory` similar to https://github.com/Azure/azure-functions-language-worker-protobuf/blob/dev/src/proto/FunctionRpc.proto#L200

Reply:
Yes, this is how host indicates to the worker that it needs the metadata. I can rename it, that will definitely add more context and consistency!